iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesAnalyzer.java

```diff
@@ -78,6 +80,19 @@ public Tree parse(String content, @Nullable InputFileContext inputFileContext) {
     }
   }
 
+  @Override
+  protected FileWithAsts fileWithAsts(InputFileContext inputFileContext, Tree tree) {
+    Tree additionalTree = null;
+    if (tree instanceof HelmFileTreeImpl helmFileTree) {
+      additionalTree = helmFileTree.getGoTemplateAst();
+    }
+    if (additionalTree != null) {
+      return new FileWithAsts(inputFileContext, tree, Set.of(additionalTree));
+    } else {
+      return super.fileWithAsts(inputFileContext, tree);
+    }
+  }
+
```

Comment:
Looks a bit strange, I would change the code a bit:
```suggestion
  @Override
  protected FileWithAsts fileWithAsts(InputFileContext inputFileContext, Tree tree) {
    if (tree instanceof HelmFileTreeImpl helmFileTree) {
      Tree additionalTree = helmFileTree.getGoTemplateAst();
      return new FileWithAsts(inputFileContext, tree, Set.of(additionalTree));
    } else {
      return super.fileWithAsts(inputFileContext, tree);
    }
  }

```